### PR TITLE
Docker provision & conf. documentation

### PR DESCRIPTION
**Pull request by dLobatog**
Needs to be polished up, with screenshots, etc...

### CODE REVIEW

**Comment by mmoll** [comment]
what's the status here?

**Comment by dLobatog** [comment]
@mmoll Text is now 100% ready, missing screenshots.

File: plugins/foreman_docker/1.x/index.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+    Here, [fqdn] stands for the fully qualified domain name of your Foreman server. Skip this step if the container host is already registered as a Foreman host.
+
+3. Depending on the location of the container host, perform the following tasks:
+    * If the container host is on the same machine as the Foreman server, create a docker user group and add the foreman user to it:
+
+          # groupadd docker
+          # usermod -aG docker foreman
+
+    * Modify the OPTIONS variable in the /etc/sysconfig/docker file as follows:
```
Comment (domcleal) [inline plugins/foreman_docker/1.x/index.md]: EL and Fedora only?  If so, either cover Debian too or state it is.

File: plugins/foreman_docker/1.x/index.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+          OPTIONS='--selinux-enabled -G docker'
+
+    * Restart the affected services to apply the changes:
+
+          # systemctl restart docker.service
```
Comment (domcleal) [inline plugins/foreman_docker/1.x/index.md]: Only applicable on EL7 and Jessie?

File: plugins/foreman_docker/1.x/index.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+          # katello-service restart
+
+    * If the container host is on a different machine than the Foreman server, open a port on the container host to communicate with the Foreman server. To do so, modify the OPTIONS variable in the /etc/sysconfig/docker file as follows:
+
+          OPTIONS='--selinux-enabled -H tcp://0.0.0.0:[PORT_NUMBER] -H unix:///var/run/docker.sock'
+
+    * Replace [PORT_NUMBER] with a selected port number. Restart the docker service and verify your settings as follows:
+
+          # systemctl restart docker.service
+          # systemctl status docker.service
```
Comment (domcleal) [inline plugins/foreman_docker/1.x/index.md]: ditto

File: plugins/foreman_docker/1.x/index.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+1. Navigate to Containers → All Containers to view the list of available containers.
+2. Click the name of the container you want to delete.
+3. Click Delete.
+4. In the alert box, click OK to remove the container.
+⁠
+
+## 1.2. Working with repositories
+Follow the Katello [Docker documentation](http://www.katello.org/docs/2.2/user_guide/docker/docker.html).
+
+## 2. Help
```
Comment (domcleal) [inline plugins/foreman_docker/1.x/index.md]: Should be one # for top-level headings, ditto below.

File: plugins/foreman_docker/1.x/index.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+**Procedure 1.7. To remove a container:**
+
+1. Navigate to Containers → All Containers to view the list of available containers.
+2. Click the name of the container you want to delete.
+3. Click Delete.
+4. In the alert box, click OK to remove the container.
+⁠
+
+## 1.2. Working with repositories
+Follow the Katello [Docker documentation](http://www.katello.org/docs/2.2/user_guide/docker/docker.html).
```
Comment (domcleal) [inline plugins/foreman_docker/1.x/index.md]: 2.2?  Is there a better link?

Is there any non-Katello support that should be documented?

**Comment by domcleal** [comment]
Also needs linking from the documentation index.

File: plugins/foreman_docker/1.x/index.md [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+1. Navigate to Containers → All Containers to view the list of available containers.
+2. Click the name of the container you want to commit.
+3. Click Commit. Foreman prompts you to provide the following information:
+    * Specify a repository name. This can be a single name or combined with the user name, for example user/my-rhel-image.
+    * Assign a tag to the image.
+    * Provide your contact information.
+    * Provide an informative comment about the image.
+4. Click Submit.
+
+![Commit a container](https://camo.githubusercontent.com/b9ed7a7db20265b8740498ee47115cdc7b27903d/687474703a2f2f692e696d6775722e636f6d2f636f463559304c2e706e67)
```
Comment (domcleal) [inline plugins/foreman_docker/1.x/index.md]: This should be in the repo.

File: plugins/foreman_docker/1.x/index.md [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+
+**Procedure 1.4. To Investigate a container:**
+
+1. Navigate to Containers → All Containers.
+2. On the Containers page, every Docker compute resource has a dedicated tab. Each of these tabs contains the table of available containers together with selected parameters of each container. Select the tab of the compute resource you want to inspect.
+3. To view the container metadata, click the name of the container you want to inspect. Foreman displays the table of container properties.
+4. On the Processes tab, you can view processes that are currently running in the container. Click on the process name to view the metadata of the process.
+5. If the container is running, you can view its standard output in the Logs tab. If you selected the allocate a pseudo-tty check box when creating a container, the console is interactive. Otherwise, it displays the initial standard output produced when the container started.
+
+![Container show page](https://camo.githubusercontent.com/f7fdb2a77bb6bc6bac713afb03ddcc7f93d8a297/687474703a2f2f692e696d6775722e636f6d2f586e7250545a432e706e67)
```
Comment (domcleal) [inline plugins/foreman_docker/1.x/index.md]: ditto

**Comment by dLobatog** [comment]
Fixed, sorry for keeping the assets out. Latest commit puts them in the static/images/plugins/foreman_docker folder.

File: plugins/foreman_docker/1.x/index.md [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+
+    # apt-get install ruby-foreman-docker
+
+### Bundle (gem)
+
+Add the following to bundler.d/Gemfile.local.rb in your Foreman installation directory (/usr/share/foreman by default)
+
+    $ gem 'foreman_docker'
+
+Then run `bundle install` and `foreman-rake db:migrate` from the same directory
```
Comment (domcleal) [inline plugins/foreman_docker/1.x/index.md]: Nitpick, this sort of muddles a source and package installation a bit, I'd avoid recommending it or prefix this section with a notice to say that it's only for _source_ installations.

(foreman-rake's a packaged command, bundle install wouldn't be recommended/wouldn't work on Debian or RPM installations.)

Comment (dLobatog) [inline plugins/foreman_docker/1.x/index.md]: Oh right, I'll change it to just 'rake' and add a small notice.

**Comment by dLobatog** [comment]
Added a notice so that the section that explains how to install foreman-docker from source explicitly recommends not to run these commands if you have installed Foreman via packages.

**Comment by domcleal** [comment]
Thanks!

**Pull request merged**